packages/node/src/process-queue.ts

```diff
@@ -1,9 +1,38 @@
 import Queue, { Task } from "p-queue";
 
+import MemoryLockService from "./lock/memory-lock-service";
+import { ILockInterface } from "./lock/types";
 import { addToManyQueues } from "./methods/queued-execution";
 
+class QueueWithLockingServiceConnection extends Queue {
+  constructor(
+    private readonly queueName,
+    private readonly lockingService: ILockInterface,
+    ...args: any[]
+  ) {
+    super(...args);
+  }
+
+  async add(task: Task<any>) {
+    return super.add(() =>
+      this.lockingService.acquireLock(this.queueName, task, 30_000)
+    );
```

Comment:
This is the important bit. It's meant to add the process to the queue, then when it gets executed locally, request the lock before continuing, otherwise error. I think this is the right order...